mongo/gridfs/bucket.go

```diff
@@ -265,12 +284,21 @@ func (b *Bucket) Delete(fileID interface{}) error {
 //
 // If this download requires a custom read deadline to be set on the bucket, it cannot be done concurrently with other
 // read operations operations on this bucket that also require a custom deadline.
+//
+// Deprecated: This method is deprecated and will eventually be removed in version 2.0 of the driver. FindContext
+// should be used instead.
 func (b *Bucket) Find(filter interface{}, opts ...*options.GridFSFindOptions) (*mongo.Cursor, error) {
```

Comment:
In the next Go Driver major version, will we update `Find` to include a `Context` argument and remove `FindContext`, or will we remove `Find` and keep `FindContext`? If it's the former, we should not mark this as deprecated but just say that it's the same as calling `FindContext` with `context.Background()` (for example, check out the description of [sql.DB.Query](https://pkg.go.dev/database/sql#DB.Query)). Same question for all of the new `*Context` functions.

Also, should we deprecate `Bucket.SetReadDeadline` and `Bucket.SetReadDeadline`?

Reply:
You may be right, but the path forward here is a little confusing. See my comment [here](https://github.com/mongodb/mongo-go-driver/pull/1016#pullrequestreview-1031871916). 

`Delete`, `Drop`, `Find` and `Rename` use the `readDeadline` or `writeDeadline` to create a context with that deadline and run the internal operations (so they do _not_ always use `context.Background()`). I think in 2.0 we should change those methods to take a context parameter and remove the `Context` versions of the methods. The stream management methods (`OpenUploadStream*`, `UploadFromStream*`, `OpenDownloadStream*` and `DownloadToStream*`) do not currently have `Context` versions and also use `readDeadline` and `writeDeadline`. 

If we want to remove `SetReadDeadline` and `SetWriteDeadline`, I think we should:
- Add `Context` versions of every `Bucket` method.
- Deprecate all `Context` methods to indicate they will be removed in version 2.0 of the driver.
- Add `NOTE`s to every non-context method to indicate they use `readDeadline` and `writeDeadline`.
- Deprecate `SetReadDeadline` and `SetWriteDeadline` methods to indicate they will be removed in version 2.0 of the driver.

Again, this is a slightly larger change, and I worry that GridFS would become highly non-intuitive with two different versions of every method. But, it is perhaps equally strange to use a combination of deadlines and contexts to define timeouts... Let me know what you think 🤷

Reply:
I am in favor of removing `SetReadDeadline` and `SetWriteDeadline` in future work.

Reply:
Sounds good. After offline discussion I have added documentation to `Delete`, `Drop`, `Find` and `Rename` to indicate they use `readDeadline` and `writeDeadline`. I've also added documentation to `DeleteContext`, `DropContext`, `FindContext` and `RenameContext` to indicate they use the provided context parameter and ignore the *Deadline variables.